src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -55,7 +64,77 @@ public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String>
    * @return a list of IexHistoricalPrices objects
    */
   public List<IexHistoricalPrices> getHistoricalPricesForSymbol(final String symbol, final String range) {
-      return iexClient.getHistoricalPricesForSymbol(symbol, range);
+
+      LocalDate sinceDate;
+      Integer rangeValue = Integer.parseInt(range.replaceAll("[^0-9]",""));
+
+      switch(range.charAt(range.length()-1)){
+          case 'd': sinceDate = LocalDate.now().minusDays(rangeValue); break;
+          case 'm': sinceDate = LocalDate.now().minusMonths(rangeValue); break;
+          case 'y': sinceDate = LocalDate.now().minusYears(rangeValue); break;
+          case 'q': sinceDate = LocalDate.now().minusMonths(rangeValue*3); break;
+          default: return new ArrayList<IexHistoricalPrices>();
```

Comment:
Is this really what you want to do here? This case would hit if a caller of this class used any invalid period specifier -- returning an empty list is misleading because the caller would reasonably interpret that as meaning there's no data for their request. In reality, their request was bad -- so this is a place we should throw an Exception. Java's built in IllegalArgumentException works fine here, no need to define a custom exception type.

Reply:
Done.